### PR TITLE
OAS-3818 | Oasis private network deployments (Azure Private Link)

### PR DESCRIPTION
**Pull request by Simran-B**
Adds section about creating a private network deployment.

https://deploy-preview-896--zealous-morse-14392b.netlify.app/docs/3.8/oasis/deployments.html#how-to-create-a-private-network-deployment

I haven't added all of the screenshots I took but I can add them if desired. Couldn't document the Azure alias dialog because there were errors when I tested the feature.

### CODE REVIEW

File: oasis/deployments.md [hunk 1]
```diff
@@ -166,6 +166,47 @@ and automatically accepted by most web browsers. The certificate used on port
...
+
+The deployment will not be reachable from public internet anymore, other than via
+the Oasis dashboard to administrate it. To revert to a public deployment, please
+contact support via _Request help_ in the help menu.
+
+1. **View** the deployment you want to change.
+2. On the **Overview** tab, click the **Edit** button with an ellipsis (`…`)
+   icon. If you see a pencil icon and no menu opens, then you are either on the
+   free-to-try tier or the selected deployment is not eligible. The private
+   network option in only available on the professional tier, for OneShard and
```
Comment (Robert-Stam) [inline oasis/deployments.md:190]: professional  ---> enterprise

File: oasis/deployments.md [hunk 2]
```diff
@@ -166,6 +166,47 @@ and automatically accepted by most web browsers. The certificate used on port
...
+to a private network deployment afterwards.
+
+The deployment will not be reachable from public internet anymore, other than via
+the Oasis dashboard to administrate it. To revert to a public deployment, please
+contact support via _Request help_ in the help menu.
+
+1. **View** the deployment you want to change.
+2. On the **Overview** tab, click the **Edit** button with an ellipsis (`…`)
+   icon. If you see a pencil icon and no menu opens, then you are either on the
+   free-to-try tier or the selected deployment is not eligible. The private
```
Comment (Robert-Stam) [inline oasis/deployments.md:189]: free-to-try or professional

File: oasis/deployments.md [hunk 3]
```diff
@@ -166,6 +166,47 @@ and automatically accepted by most web browsers. The certificate used on port
...
+   network option in only available on the professional tier, for OneShard and
+   Sharded deployments running on Azure.
+3. Click **Change to private network** in the menu
+   ![Oasis Deployment Private Network Menu](images/oasis-deployment-private-network-menu.png)
+4. A dialog opens for the configuration. Read the explanation and click **Next**.
+5. You need to enter one or more Azure Subscription IDs (UUIDs). They cannot be
+   changed later. Proceed by clicking **Next**.
```
Comment (Robert-Stam) [inline oasis/deployments.md:196]: They can be changed until the first connection has been established

**Review by Robert-Stam — APPROVED** [review]
See remarks

File: oasis/deployments.md [hunk 4]
```diff
@@ -166,6 +166,47 @@ and automatically accepted by most web browsers. The certificate used on port
...
+   They can be changed later.
+7. Click **Confirm Settings** to change the deployment.
+8. Back on the **Overview** tab, scroll down to the **Private Network** section
+   that is now displayed to see the connection status and to change the
+   configuration.
+   ![Oasis Deployment Private Network](images/oasis-deployment-private-network.png)
+9. Oasis will configure a Private Endpoint Service. As soon as the **Azure alias**
+   becomes available, you can copy it and then go to your Microsoft Azure portal
+   to create Private Endpoints using this alias.
+
```
Comment (nikhil-varma) [inline oasis/deployments.md:209]: I think we can also add that the connection details will be visible post it is established from the Portal? 

This will show the number of connections established 
image

File: oasis/deployments.md [hunk 5]
```diff
@@ -166,6 +166,47 @@ and automatically accepted by most web browsers. The certificate used on port
...
+1. **View** the deployment you want to change.
+2. On the **Overview** tab, click the **Edit** button with an ellipsis (`…`)
+   icon. If you see a pencil icon and no menu opens, then you are either on the
+   free-to-try tier or the selected deployment is not eligible. The private
+   network option in only available on the professional tier, for OneShard and
+   Sharded deployments running on Azure.
+3. Click **Change to private network** in the menu
+   ![Oasis Deployment Private Network Menu](images/oasis-deployment-private-network-menu.png)
+4. A dialog opens for the configuration. Read the explanation and click **Next**.
```
Comment (nikhil-varma) [inline oasis/deployments.md:194]: Maybe we can call this a wizard instead of a dialog since it takes the users' through different steps?

File: oasis/deployments.md [hunk 6]
```diff
@@ -166,6 +166,47 @@ and automatically accepted by most web browsers. The certificate used on port
...
+5. You need to enter one or more Azure Subscription IDs (UUIDs). They cannot be
```
Comment (nikhil-varma) [inline oasis/deployments.md:195]: Isn't this supposed to be GUIDs @Robert-Stam ?

File: oasis/deployments.md (hunk 5 above)
Comment (ansoboleva) [reply oasis/deployments.md:194]: Second that. Would rephrase altogether as "In the Configuration wizard, click Next to enter your configuration details."

File: oasis/deployments.md [hunk 7]
```diff
@@ -166,6 +166,50 @@ and automatically accepted by most web browsers. The certificate used on port
...
+
+The deployment will not be reachable from public internet anymore, other than via
+the Oasis dashboard to administrate it. To revert to a public deployment, please
+contact support via _Request help_ in the help menu.
+
+1. **View** the deployment you want to change.
+2. On the **Overview** tab, click the **Edit** button with an ellipsis (`…`)
+   icon. If you see a pencil icon and no menu opens, then you are either on the
+   free-to-try or professional tier, or the selected deployment is not eligible.
+   The private network option in only available on the enterprise tier, for
```
Comment (Robert-Stam) [inline oasis/deployments.md:190]: in --> is

**Review by Robert-Stam — COMMENTED** [review]
LGTM, but 1 typo

**Review by ansoboleva — APPROVED** [review]
reviewed
